src/agent/LibFuzzerDotnetLoader/README.md

```diff
@@ -0,0 +1,134 @@
+# LibFuzzerDotnetLoader
+
+## About
+
+`LibFuzzerDotnetLoader` is a dynamic, reusable SharpFuzz harness for use with
+[SharpFuzz](https://github.com/Metalnem/sharpfuzz) with [libfuzzer-dotnet](https://github.com/Metalnem/libfuzzer-dotnet).
+It removes the need for your project to link against SharpFuzz and invoke `SharpFuzz.LibFuzzer.LibFuzzerDotnetLoader`.
+
+## Usage
+
+Suppose your project ships a managed library, `Fabrikam.dll`,
+and we want to fuzz the `AppModel.Deserialize()` method below:
+
+```csharp
+namespace Fabrikam {
+    class AppModel
+    {
+        // ...
+
+        public static AppModel Deserialize(ReadOnlySpan<byte> data)
+        {
+            // ..
+        }
+    }
+}
+```
+
+We need export a static _test method_ with the signature `static void (ReadOnlySpan<byte>)`.
+If your project does not support `Span` types,
+you may instead export a method with the signature `static void (byte[])`,
+with a performance cost of one extra copy of the input for each fuzzing iteration.
+
+Our project may then look like:
+
+```csharp
+namespace Fabrikam {
+    class AppModel
+    {
+        // ...
+
+        public static AppModel Deserialize(ReadOnlySpan<byte> data)
+        {
+            // ..
+        }
+    }
+
+    class AppModelFuzzer
+    {
+        // The test method, invoked repeatedly when fuzzing.
+        //
+        // How we name this doesn't matter: we only require a compatible signature.
+        static void TestInput(ReadOnlySpan<byte> data)
+        {
+            // In this case, we just ignore successful results. Sometimes it may make
+            // sense to make assertions about the result, test serialization/deserialization
+            // round-tripping, etc.
+            //
+            // Any uncaught exception will be treated as a fuzzing test case failure.
+            AppModel.Deserialize(data);
+        }
+    }
+}
+```
+
+Assuming you have compiled `libfuzzer-dotnet.exe` and `LibFuzzerDotnetLoader` for your platform (see below),
+you can now fuzz using the following steps:
+
+```pwsh
+# Create a corpus directory to store fuzzer-generated inputs that uncover new blocks of code.
+mkdir corpus
+
+# Instrument the DLL in-place to provide coverage feedback to guide fuzzing.
+# This makes fuzzing much more effective.
+#
+# This only needs to be done (exactly) once for each DLL under test.
```

Comment:
if you do it more than once does it break things?

Reply:
Nope, invocations of the `sharpfuzz` tool are idempotent.

Reply:
(this is clear from the error message if you attempt it, too)